kernel.py

```diff
@@ -211,6 +232,8 @@ def _start_kernel(self):
         self._create_kernel()
 
         self._logger.info("IPython: Start kernel now. pid: %i, thread: %r" % (os.getpid(), threading.current_thread()))
+        self._init_io()
```

Comment:
I don't think it's a good idea to do that by default. The intention of this project (background-zmq-ipython) is that it runs in the background. It should not change and affect the main program (like changing `sys.stdout`).

But I see your use case for that. So better just make that an option for `IPythonBackgroundKernelWrapper`.

Reply:
Fix: 54970642 but not a parameter yet (waiting for your opinion)

Reply:
This is somewhat hacky, and could have unintended side effects. I really would make this only optional. Some option like `redirect_stdio` which by default is `False`, and if `True`, it uses your code.

Reply:
Also, I'm actually not sure what the nicer behavior is. Maybe for `redirect_stdio=True`, better really redirect all stdio, not just from this thread. Or maybe only if `redirect_stdio="this_thread_only"` or so, not sure. I would leave that away for now, and either redirect it not all all, or fully.

Reply:
Made it optional (default disabled) : 37c526d
If enabled, it redirects the current thread <-  I am personally interested in only the thread output (the `print` from the controlling console).
I tested with different client and it behaves exactly as I need with `redirect_stdio=True`.

By  the way I agree that the default ipykernel implementation "should" look like yours. But that won't be the case for years. So ... thanks again for your code.

Reply:
Ok.